src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -38,38 +39,67 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
+        /// Rewrites the given interpolated string to the set of handler creation and Append calls, returning an array builder of the append calls and the result
         /// local temp.
         /// </summary>
         /// <remarks>Caller is responsible for freeing the ArrayBuilder</remarks>
-        private (ArrayBuilder<BoundExpression> BuilderPatternExpressions, BoundLocal Result) RewriteToInterpolatedStringHandlerPattern(BoundInterpolatedString node)
+        private (ArrayBuilder<BoundExpression> HandlerPatternExpressions, BoundLocal Result) RewriteToInterpolatedStringHandlerPattern(BoundInterpolatedString node)
         {
             Debug.Assert(node.InterpolationData is { Construction: not null });
             Debug.Assert(node.Parts.All(static p => p is BoundCall or BoundStringInsert { Value: BoundCall }));
             var data = node.InterpolationData.Value;
             var builderTempSymbol = _factory.InterpolatedStringHandlerLocal(data.BuilderType, data.ScopeOfContainingExpression, node.Syntax);
-            var builderTemp = _factory.Local(builderTempSymbol);
+            BoundLocal builderTemp = _factory.Local(builderTempSymbol);
 
             // PROTOTYPE(interp-string): Support dynamic creation
-            // var builder = new BuilderType(baseStringLength, numFormatHoles);
-            Debug.Assert(data.Construction is BoundObjectCreationExpression);
-            var builderConstruction = _factory.AssignmentExpression(builderTemp, (BoundExpression)VisitObjectCreationExpression((BoundObjectCreationExpression)data.Construction));
+            // var handler = new HandlerType(baseStringLength, numFormatHoles, ...InterpolatedStringHandlerArgumentAttribute parameters, <optional> out bool handlerIsValid);
+            var construction = (BoundObjectCreationExpression)data.Construction;
+
+            BoundLocal? appendShouldProceedLocal = null;
+            bool ownsAppendShouldProceedLocal = false;
+            if (data.HasTrailingHandlerValidityParameter)
+            {
+                Debug.Assert(construction.ArgumentRefKindsOpt[^1] == RefKind.Out);
+
+                BoundInterpolatedStringArgumentPlaceholder trailingParameter = data.ArgumentPlaceholders[^1];
+                if (!HasPlaceholderReplacement(trailingParameter))
+                {
+                    ownsAppendShouldProceedLocal = true;
+                    TypeSymbol localType = trailingParameter.Type!;
+                    Debug.Assert(localType.SpecialType == SpecialType.System_Boolean);
+                    var outLocal = _factory.SynthesizedLocal(localType);
+                    appendShouldProceedLocal = _factory.Local(outLocal);
+
+                    AddPlaceholderReplacement(trailingParameter, appendShouldProceedLocal);
+                }
+                else
+                {
+                    // We're already processing this node, so use the local VisitArguments will have allocated for us
```

Comment:
>// We're already processing this node, so use the local VisitArguments will have allocated for us [](http://example.com/codeflow?start=20&length=97)

I am having trouble coming up with a scenario that will have this "re-entrancy". Could you elaborate please?